### PR TITLE
Mozilla bug 1861068 - Make nsHTML5AttributeName constructors not allocate

### PR DESCRIPTION
**Pull request by emilio**

### CODE REVIEW

**Review by hsivonen — CHANGES_REQUESTED** [review]
The fourth name (`l3`/`htmlLang`) should not show up on the C++ side.

File: src/nu/validator/htmlparser/impl/AttributeName.java [hunk 1]
```diff
@@ -424,25 +338,32 @@ public final class AttributeName
      *            whether this is an xmlns attribute
      */
     private AttributeName(@NsUri @NoLength String[] uri,
-            @StaticLocal @NoLength String[] local, @Prefix @NoLength String[] prefix
+            @StaticLocal String l0, @StaticLocal String l1, @StaticLocal String l2, @StaticLocal String l3, @Prefix @NoLength String[] prefix
```
[the remark this answers is not in the document]
Comment (emilio) [reply src/nu/validator/htmlparser/impl/AttributeName.java]: Done.

File: translator-src/nu/validator/htmlparser/cpptranslate/CppVisitor.java [hunk 2]
```diff
@@ -1444,8 +1444,9 @@ public void visit(ObjectCreationExpr n, LocalSymbolTable arg) {
 
         if ("AttributeName".equals(n.getType().getName())) {
             List<Expression> args = n.getArgs();
-            while (args != null && args.size() > 3) {
-                args.remove(3);
+            if (args != null) {
+                // Remove flags from AttributeName constructor.
+                args.remove(args.size() - 1);
```
[the remark this answers is not in the document]
Comment (emilio) [reply translator-src/nu/validator/htmlparser/cpptranslate/CppVisitor.java]: Both, right? Both are unused in C++.

**Review by hsivonen — APPROVED** [review]
Thanks!

(I'll merge after the Gecko change lands.)

**Pull request merged**